worldcup/README.md

```diff
@@ -22,21 +22,111 @@ Mexico: 0.1% chance of winning
+  
 
 ## Implementation Details
 
-TODO
+* Complete the implementation of `volume.c` at right, such that it simulates a number of tournaments and outputs each team's probability of winning, for all probabilities greater than `0%`.
+
+* In `main()`, your program should read in the file indicated by the user into the list `bracket`. This list should be a list of dictionaries, each dictionary containing `team` and `rating` keys, where the `rating` is a numerical value. 
+
+  * Two files are provided: `2018m.csv` and `2019w.csv`. 
+
+  * In each, the bracket of teams, in order, is provided along with their FIFA Ratings. 
+    
+    * Specifically, in the first round, the team in line 1 will play the team in line 2. The team in line 3 will play the team in line 4. Since there are 16 total lines, the first round will contain 8 total games. 
+
+    * In the second round, the winner between lines 1 and 2 will play the winner between lines 3 and 4, etc.
+
+    * The first few lines in `2018m.csv` look like this: 
+    
+    ```
+      team,rating
+      Uruguay,976
+      Portugal,1306
+      France,1166
+    ```
+
+* After, in `main()`, your program should simulate `N` tournaments, which, notice, has already been defined as `1000`. For each tournament, your program should return the winner and keep track of the number of wins for each team in the dictionary `counts`.
+
+* In `simulate_tournament(bracket)`, your program should return the name of the winning team of one simulation of the given bracket. Your program should iterate through as many rounds as necessary to output one single winner of the tournament. 
+
+* In `simulate_round(bracket)`, your program should simulate a single round of the tournament and return the winners of each game in that round.  
+  
+  * Note that in the first round, the first team should play the second, the third team should play the fourth, and so on. 
+
+  * Additionally, for example, if the bracket given has 4 teams (third round), your program should simulate 2 games and return a list of the 2 winning teams. 
+
+* Notice that `simulate_game(team1, team2)` has already been completed. Given two teams, it calculates the probability the first team defeats the second and returns `True` with that probability and `False` otherwise. We return `random.random() < probability` in this function, which is an application of Python's `random` module, where `random.random()` returns a pseudo-random value between 0 and 1. 
+
 
 ### Hints
 
-TODO
+* When reading in the file, you may find this syntax helpful, with `file_name` as the name of your file and `var` as a variable.
 
+    ```python
+    with open(file_name) as var:
+        reader = csv.DictReader(var)
+    ```
+
+* Recall that a dictionary looks like this: `[('team', 'Uruguay'), ('rating', 976)]`. If this dictionary had variable name `example`, then `example["team"] = Uruguay` and `example["rating"] = 976`. 
```

Comment:
`[('team', 'Uruguay'), ('rating', 976)]` would technically be interpreted as a list of tuples, even though Python's ordered dict will display its contents this way. For consistency with lectures, though, probably best to go with `{"team": "Uruguay", "rating": "976"}`.